code/modules/mob/living/carbon/human/species_types/jellypeople.dm

```diff
@@ -668,35 +673,35 @@
 	background_icon_state = "bg_alien"
 
 /datum/action/innate/linked_speech/Activate()
-	var/mob/living/carbon/human/H = owner
-	if(H.stat == DEAD)
+	var/mob/living/carbon/human/human_user = owner
+	if(human_user.stat == DEAD)
 		return
-	var/datum/species/jelly/stargazer/species = target
-	if(!species || !(H in species.linked_mobs))
-		to_chat(H, span_warning("The link seems to have been severed..."))
-		Remove(H)
+
+	var/datum/species/jelly/stargazer/master_species = target
```

Comment:
target is supposed to be an /obj type, could you change this to be reflected in code?